ssl/d1_lib.c

```diff
@@ -485,9 +645,16 @@ int DTLSv1_listen(SSL *ssl, BIO_ADDR *client)
         OPENSSL_free(buf);
         return -1;
     }
+    cookie = OPENSSL_malloc(65535); // Todo: magic number, haven't found a definition in the source code yet.
+    if (cookie == NULL) {
+        OPENSSL_free(buf);
+        OPENSSL_free(wbuf);
+        return -1;
+    }
```

Comment:
Why is this change needed, especially why the fairly large allocation size?

Reply:
My rationale was that since RFC 8446 section 4.2.2 defines the cookie as:

>      struct {
>          opaque cookie<1..2^16-1>;
>       } Cookie;

This code should support the larger size, but maybe it was never the intention for this code? Any case I've reverted the code to the old functionality.

Reply:
I am actually not sure about that. Perhaps @mattcaswell has an opinion?